src/Nri/Ui/Button/V9.elm

```diff
@@ -0,0 +1,887 @@
+module Nri.Ui.Button.V9 exposing
+    ( button, link
+    , Attribute
+    , label, icon, custom
+    , onClick, href
+    , linkSpa, linkExternal, linkWithMethod, linkWithTracking, linkExternalWithTracking
+    , small, medium, large
+    , exactWidth, unboundedWidth, fillContainerWidth
+    , primary, secondary, danger, premium
+    , enabled, unfulfilled, disabled, error, loading, success
+    , delete
+    , toggleButton
+    )
+
+{-|
+
+
+# Changes from V8:
+
+  - Changes API to be attribute-based, rather than config-based
+
+
+# Create a button or link
+
+@docs button, link
+@docs Attribute
+@docs label, icon, custom
+
+
+## Add event listeners and hrefs
+
+@docs onClick, href
+@docs linkSpa, linkExternal, linkWithMethod, linkWithTracking, linkExternalWithTracking
+
+
+## Sizing
+
+@docs small, medium, large
+@docs exactWidth, unboundedWidth, fillContainerWidth
+
+
+## Change the color scheme
+
+@docs primary, secondary, danger, premium
+
+
+## Change the state (buttons only)
+
+@docs enabled, unfulfilled, disabled, error, loading, success
+
+
+# Commonly-used buttons
+
+@docs delete
+@docs toggleButton
+
+-}
+
+import Accessibility.Styled as Html exposing (Attribute, Html)
+import Accessibility.Styled.Role as Role
+import Accessibility.Styled.Widget as Widget
+import Css exposing (Style)
+import Css.Global
+import EventExtras.Styled as EventExtras
+import Html as RootHtml
+import Html.Styled as Styled
+import Html.Styled.Attributes as Attributes
+import Html.Styled.Events as Events
+import Json.Decode
+import Markdown.Block
+import Markdown.Inline
+import Nri.Ui
+import Nri.Ui.AssetPath as AssetPath exposing (Asset)
+import Nri.Ui.Colors.Extra as ColorsExtra
+import Nri.Ui.Colors.V1 as Colors
+import Nri.Ui.Fonts.V1
+import Nri.Ui.Html.Attributes.V2 as AttributesExtra
+import Nri.Ui.Svg.V1 as NriSvg exposing (Svg)
+import Svg
+import Svg.Attributes
+
+
+styledName : String -> String
+styledName suffix =
+    "Nri-Ui-Button-V9-" ++ suffix
+
+
+{-|
+
+    Button.button
+        [ Button.label "My great button!"
+        , Button.onClick ()
+        , Button.enabled
+        ]
+
+-}
+button : List (Attribute msg) -> Html msg
+button attributes =
+    List.foldl (\attribute b -> attribute b) build attributes
+        |> renderButton
+
+
+{-|
+
+    Button.link
+        [ Button.label "My great link!"
+        , Button.href "My href"
+        , Button.secondary
+        ]
+
+-}
+link : List (Attribute msg) -> Html msg
+link attributes =
+    List.foldl (\attribute l -> attribute l) build attributes
+        |> renderLink
+
+
+{-| -}
+label : String -> Attribute msg
+label label_ (ButtonOrLink config) =
+    ButtonOrLink { config | label = label_ }
+
+
+{-| -}
+icon : Svg -> Attribute msg
```

Comment:
💭 I think we'd want to enforce that icon buttons always also have a label to use for alt text, so maybe icon should be `icon : String -> Svg -> Attribute msg` and set both the label and the icon at once?

Reply:
I think actually we don't want to label icon -- the icon's purpose is descriptive, and when that's the case the alt text should be explicitly set to `""`. 

That said, I don't have any idea about how to enforce that  here 🤔

Reply:
Oh, right, there's still text when there's an icon.  That makes sense!

Hmm, maybe label actually shouldn't be an optional Attribute, then, and should just be a required string parameter to the button functions?

Reply:
An idea that occurred to me is to let the button take html children, like a normal button:

```
Button.button 
    [  Button.primary
    ,  Button.onClick DoAThing
    ]
    [  Html.text "My label"
    ]
```

Or maybe:

```
Button.button 
    [  Button.primary
    ,  Button.onClick DoAThing
    ]
    [  Button.icon myIcon
    ,  Button.label "HELLO!!!"
    ]
```

Reply:
Yeah I think required string sounds good..

Reply:
I'd be hesitant about having it take a list of children because I suspect we didn't really think through making the CSS handling arbitrary children -- things like wrapping, vertical alignment, containing inline vs inline-block vs block elements, dealing with things of unexpected height, etc would all need to be handled if we allowed more flexibility in specifying the children.